custom/lib_portal/models/product_template.py

```diff
@@ -3,17 +3,29 @@
 class ProductTemplate(models.Model):
     _inherit = 'product.template'
     
-    is_book = fields.Boolean(string='Book?', default=False)
+    is_book = fields.Boolean(string='Is a Book', default=False, store=True)
     author_id = fields.Many2one('res.partner', 
                               string='Select Author', 
                               domain=[('is_company', '=', False)])
     author_name = fields.Char(related='author_id.name', store=True)
-    book_description = fields.Text(string='Book Description')
+    description = fields.Text(string='Book Description')
     categ_id = fields.Many2one('product.category', string='Book Genre')
     publication_date = fields.Date(string='Publication Date')
     book_language = fields.Many2one('res.lang', string='Book Language')
     default_code = fields.Char(string='Shelf Location')
     barcode = fields.Char(string='Book ISBN')
+    publisher_id = fields.Many2one('res.partner', 
+                              string='Publisher', 
+                              domain=['&',('is_company', '=', True),
+                                      ('is_publisher', '=', True)])
+    quantity_id = fields.Many2one('stock.quant')
+    qty_available = fields.Float(related='quantity_id.available_quantity', store=True)
```

Comment:
odoo already provides a mechanism to know the availability of the products

Reply:
Yeah I agree.This was developed in the previous week tasks.This is to be modified in the current task to check the availability from existing stock picking module